cmd/kubectl-gadget/utils/trace.go

```diff
@@ -132,42 +131,31 @@ func printTraceFeedback(m map[string]string, totalNodes int) {
 	}
 }
 
-func deleteTraces(traceRestClient *restclient.RESTClient, traceID string) {
+func deleteTraces(traceClient *clientset.Clientset, traceID string) {
 	var listTracesOptions = metav1.ListOptions{
 		LabelSelector: fmt.Sprintf("%s=%s", GLOBAL_TRACE_ID, traceID),
 	}
-	err := traceRestClient.
-		Delete().
-		Namespace("gadget").
-		Resource("traces").
-		VersionedParams(&listTracesOptions, scheme.ParameterCodec).
-		Do(context.TODO()).
-		Error()
+
+	err := traceClient.GadgetV1alpha1().Traces("gadget").DeleteCollection(
+		context.TODO(), metav1.DeleteOptions{}, listTracesOptions,
+	)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error deleting traces: %q", err)
 	}
 }
 
-// getRestClient returns the RESTClient associated with kubeRestConfig() for
-// APIPath /apis and GroupVersion &gadgetv1alpha1.GroupVersion.
-func getRestClient() (*restclient.RESTClient, error) {
-	restConfig, err := kubeRestConfig()
+func getTraceClient() (*clientset.Clientset, error) {
+	config, err := KubernetesConfigFlags.ToRESTConfig()
 	if err != nil {
-		return nil, fmt.Errorf("Error while getting rest config: %w", err)
+		return nil, err
 	}
 
-	traceConfig := *restConfig
-	traceConfig.ContentConfig.GroupVersion = &gadgetv1alpha1.GroupVersion
-	traceConfig.APIPath = "/apis"
-	traceConfig.NegotiatedSerializer = serializer.NewCodecFactory(scheme.Scheme)
-	traceConfig.UserAgent = restclient.DefaultKubernetesUserAgent()
-
-	traceRestClient, err := restclient.UnversionedRESTClientFor(&traceConfig)
+	traceClient, err := clientset.NewForConfig(config)
 	if err != nil {
-		return nil, fmt.Errorf("Error setting up trace REST client: %w", err)
+		return nil, err
 	}
```

Comment:
Why did you remove the error message this function was appending to the returned errors? Are they good enough to understand what happened? I am saying that because all the `getTraceClient` callers are forwarding the error as it is. Therefore, the error returned by this function will be directly shown to the user.

Reply:
Right. I added them back.